packages/ui/src/modal/TokenLineItem.tsx

```diff
@@ -20,13 +23,20 @@ const TokenLineItem: FC<TokenLineItemProps> = ({
   usdConversion = 0,
   isSuspicious,
   isUnavailable,
+  isOffer,
 }) => {
   const marketData = tokenDetails?.market
-  let price: number = marketData?.floorAsk?.price?.amount?.decimal || 0
+
+  let price: number = 0
   const currency = marketData?.floorAsk?.price?.currency
+  if (isOffer) {
+    price = marketData?.topBid?.price?.amount?.decimal || 0
+  } else {
+    price = marketData?.floorAsk?.price?.amount?.decimal || 0
```

Comment:
Add a prop to keep this logic clean, update the BuyModal to pass the price or the lastSell value.